### PR TITLE
Redux higher order component

### PR DESCRIPTION
**Pull request by jattoabdul**
I restructured the app to use higher order component for the dashboard components.

### CODE REVIEW

File: client/src/js/components/main/addUserToGroup.jsx [hunk 1]
```diff
@@ -72,7 +80,8 @@ class AddUserToGroupBoard extends React.Component {
     return (
         <div id="dashContainer" className="teal">
             <div id="appContainer" className="row no-marginbtm">
-            <SideMenu onShowGroupMessages={this.onShowGroupMessages}/>
+            <SideMenu
```
Comment (BolajiOlajide) [inline client/src/js/components/main/addUserToGroup.jsx]: This should be indented properly.

File: client/src/js/components/main/creategroupboard.jsx [hunk 2]
```diff
@@ -34,7 +36,7 @@ class CreateGroupBoard extends React.Component {
     }
     const groupCreateParams = `name=${this.name.value}&desc=${this.desc.value}`;
     // making calls to the create group API endpoint
-    Api(groupCreateParams, "/api/groups/", "POST").then(
+    Api(groupCreateParams, "/api/v1/groups/", "POST").then(
         (createGroupResponse) => {
           console.log(createGroupResponse);
```
Comment (BolajiOlajide) [inline client/src/js/components/main/creategroupboard.jsx]: Remove all console statements.

File: client/src/js/components/App.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import App from './main';
+
+export default { App };
```
[the remark this answers is not in the document]
Comment (jattoabdul) [reply client/src/js/components/App.js]: no, its redundant and i have removed it.

File: client/src/js/components/main/addUserToGroup.jsx [hunk 4]
```diff
@@ -32,7 +39,7 @@ class AddUserToGroupBoard extends React.Component {
       });
     }
     // console.log(searchText);
-    Api(null, `/api/groups/${id}/usersearch?search=${searchText}`, "GET").then(
+    Api(null, `/api/v1/groups/${id}/usersearch?search=${searchText}`, "GET").then(
```
[the remark this answers is not in the document]
Comment (jattoabdul) [reply client/src/js/components/main/addUserToGroup.jsx]: this files are no longer being used. redundant and removed.

**Pull request closed without merging**